Trimble.Modus.Components/Controls/Button/TMButton.xaml.cs

```diff
@@ -1,367 +1,367 @@
+                    customButton.HeightRequest = 48;
+                    break;
+                case Enums.Size.Large:
+                    customButton.buttonLabel.FontSize = (double)Enums.FontSize.Large;
+                    customButton.buttonStackLayout.Padding = new Thickness(16, 12);
+                    customButton.HeightRequest = 56;
+                    break;
+                default:
+                    break;
+
+            }
+        }
+        else
+        {
+            if (customButton.Title != null)
+            {
+                customButton.buttonStackLayout.Padding = new Thickness(16, 0);
+            }
+            else
+            {
+                customButton.buttonStackLayout.Padding = new Thickness(16);
+            }
+        }
+
+    }
+
+    private static void UpdateFillStyleColors(TMButton customButton)
+    {
+        switch (customButton.ButtonColor)
+        {
+            case ButtonColor.Secondary:
+                customButton.buttonFrame.BackgroundColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.SecondaryButton);
+                customButton.buttonFrame.Stroke = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue);
+                customButton.buttonLabel.TextColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.White);
+
+                break;
+
+            case ButtonColor.Tertiary:
+                customButton.buttonFrame.BackgroundColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.TertiaryButton);
+                customButton.buttonFrame.Stroke = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.Transparent);
+                customButton.buttonLabel.TextColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleGray);
+                break;
+
+            case ButtonColor.Danger:
+                customButton.buttonFrame.BackgroundColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.DangerRed);
+                customButton.buttonFrame.Stroke = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.Transparent);
+                customButton.buttonLabel.TextColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.White);
+                break;
+            default:
+                customButton.buttonFrame.BackgroundColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue);
+                customButton.buttonFrame.Stroke = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.TrimbleBlue);
+                customButton.buttonLabel.TextColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.White);
+                break;
+        }
+    }
+
+    private static void UpdateOutlineStyleColors(TMButton customButton)
+    {
+        customButton.buttonFrame.BackgroundColor = (Color)ResourcesDictionary.ColorsDictionary(ColorsConstants.Transparent);
```

Comment:
This method will return color only so casting not required

Reply:
Removed Casting